### ISSUE TITLE
MultiSelect: Select all button crash browser when selected using VirtualScroll

### ISSUE THREAD

**Issue by donSan4o** [primefaces/primereact#7479]
### Describe the bug

When i try to click on select all button in virtual scroll case, it freezes my browser

### Reproducer

https://primereact.org/multiselect/#virtualscroll

### System Information

```Shell
primereact 10.8.5
```

**Comment by jerlam06** [comment]
I can confirm, this is a regression because it used to work before. My guess is that the props 'selectAll' is not being read correctly and the component actually changes all the 100k items to selected, which crashes the app. 
Sadly, we use primereact in prod and this issue makes the multiselect unusable. I thought the components were tested before deploying a new version?

**Comment by melloware** [comment]
agreed this is a bug.  I am not sure what changed but probably need to look through commits and see where it broke maybe?

**Comment by jerlam06** [comment]
@melloware After further investigation, it is broken from the version 10.8.3 onwards.

**Comment by melloware** [comment]
OK so check commits between 10.8.2 and 10.8.3 and see if you spot anything?

**Comment by sahilrajthapa** [comment]
@melloware  this issue is caused by the [line](https://github.com/primefaces/primereact/blob/master/components/lib/multiselect/MultiSelect.js#L1137).

`getInputValue` function loops through the selected values (`props.values`) and calls `getLabelByValue` for each value. 
`getLabelByValue` function, in turn, calls `findOptionByValue`, which loops through options (`props.options`) to find the corresponding option for each value.

So, this nested loop causes the performance issue.  In the worst case, when all options are selected in a large dataset (like 100K items), this will lead to 100K * 100K = 10B operations.

We can fix this issue and the [issue](https://github.com/primefaces/primereact/issues/7187) for which `getInputValue` was added by using  `value: getLabel()`. `getLabel` function already handles retrieving the selected labels, which can be passed to the input value. It also utilises `props.maxSelectedLabels` to limit these expensive operations.

If it's look good to you, I'll create a PR for this.

**Comment by melloware** [comment]
Sure create a PR I can review.

**Comment by melloware** [comment]
Thanks @sahilrajthapa !
